### PR TITLE
Add support for local embeddings for chroma

### PR DESCRIPTION
**Pull request by prasadshreyas**

### CODE REVIEW

File: dspy/retrieve/chromadb_rm.py [hunk 1]
```diff
@@ -21,12 +25,19 @@
 except ImportError:
     chromadb = None
 
+try:
+    import torch
```
Comment (okhat) [inline dspy/retrieve/chromadb_rm.py:29]: please don't import dependencies that are not in requirements.txt in the top level

you can import them inside the method you need, so people who don't need them don't see errors for them

**Comment by insop** [comment]
It seems like a good change after addressing @okhat 's feedback.

**Comment by insop** [comment]
I think with a new PR on this RM, https://github.com/stanfordnlp/dspy/pull/400, this PR might need to be revisted.

**Comment by insop** [comment]
@prasadshreyas 

Since https://github.com/stanfordnlp/dspy/pull/400 is merged, could you rebase to the main and see?

File: dspy/retrieve/chromadb_rm.py [hunk 2]
```diff
@@ -85,6 +97,20 @@ def __init__(
...
+            self._local_embed_model = AutoModel.from_pretrained(local_embed_model)
+            self._local_tokenizer = AutoTokenizer.from_pretrained(local_embed_model)
+            self.use_local_model = True
+            self.device = torch.device(
+                "cuda:0"
+                if torch.cuda.is_available()
+                else "mps"
+                if torch.backends.mps.is_available()
+                else "cpu"
+            )
```
Comment (tomaarsen) [inline dspy/retrieve/chromadb_rm.py:110]:
```suggestion
            try:
                from sentence_transformers import SentenceTransformer
            except ImportError as e:
                raise ImportError(
                    "You need to install sentence_transformers library to use pretrained embedders. "
                    "Please check the official doc https://www.sbert.net/ "
                    "or simply run `pip install sentence-transformers"
                )
            self._local_embed_model = SentenceTransformer(local_embed_model)
```

**Review by tomaarsen — COMMENTED** [review]
Additionally, I would not recommend a pure `transformers` implementation, this ends up being quite limited in what models it can support. For example, it won't work with models that use e.g. CLS pooling or extra Dense layers. Instead, you can load the model using the recommended approach, i.e. via Sentence Transformers.

**Pull request closed without merging**